test/sequential/rolldown.withdraw.fees.test.ts

```diff
@@ -42,93 +40,151 @@ beforeAll(async () => {
-  chain = "Ethereum";
 });
 
 beforeEach(async () => {
   [testUser] = setupUsers();
 });
 
 test("GIVEN a withdrawal, WHEN paying with GASP and withdrawing GASP, some fees goes to treasury", async () => {
-  const treasuryBalanceBefore = await api.query.tokens.accounts(
-    treasuryAccount,
-    GASP_ASSET_ID,
+  const treasuryAccount = "0x6d6f646c70792f74727372790000000000000000";
+  const chain = "Ethereum";
+  const treasuryBalanceBefore = stringToBN(
+    (await getTokensAccountInfo(treasuryAccount, GASP_ASSET_ID)).free,
   );
   await Sudo.batchAsSudoFinalized(Assets.mintNative(testUser));
+  const userBalanceBefore = stringToBN(
+    (await getTokensAccountInfo(testUser.keyRingPair.address, GASP_ASSET_ID))
+      .free,
+  );
+
   const events = await signTx(
     getApi(),
-    await Withdraw(testUser, withdrawalAmount, gaspIdL1Asset.ethereum, chain),
+    await Withdraw(testUser, BN_MILLION, gaspIdL1Asset.ethereum, chain),
     testUser.keyRingPair,
   );
-  const eventFiltered = filterZeroEventData(events, "Transfer");
-  const transferAmountBefore = eventFiltered.amount.replaceAll(",", "");
-  const transferAmount = new BN(transferAmountBefore);
-  expect(eventFiltered.from).toEqual(testUser.keyRingPair.address);
-  expect(eventFiltered.to).toEqual(treasuryAccount);
-  const treasuryBalanceAfter = await api.query.tokens.accounts(
-    treasuryAccount,
-    GASP_ASSET_ID,
+  const eventResponse = getEventResultFromMangataTx(events);
+  expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+
+  const transferEvent = filterAndStringifyFirstEvent(events, "Transfer");
+  const withdrawalFee = stringToBN(transferEvent.amount);
+  expect(transferEvent.from).toEqual(testUser.keyRingPair.address);
+  expect(transferEvent.to).toEqual(treasuryAccount);
+  const transactionEvent = filterAndStringifyFirstEvent(
+    events,
+    "TransactionFeePaid",
   );
-  expect(treasuryBalanceAfter.free).bnEqual(
-    treasuryBalanceBefore.free.add(transferAmount),
+  const transactionFee = stringToBN(transactionEvent.actualFee);
+
+  const treasuryBalanceAfter = stringToBN(
+    (await getTokensAccountInfo(treasuryAccount, GASP_ASSET_ID)).free,
+  );
+  const userBalanceAfter = stringToBN(
+    (await getTokensAccountInfo(testUser.keyRingPair.address, GASP_ASSET_ID))
+      .free,
+  );
+  expect(treasuryBalanceAfter).bnEqual(
+    treasuryBalanceBefore.add(withdrawalFee),
+  );
+  expect(userBalanceBefore).bnEqual(
+    userBalanceAfter.add(withdrawalFee).add(BN_MILLION).add(transactionFee),
   );
 });
 
 test("GIVEN a withdrawal, WHEN paying with GASP and withdrawing Eth, some fees goes to treasury", async () => {
-  const treasuryBalanceBefore = await api.query.tokens.accounts(
-    treasuryAccount,
-    GASP_ASSET_ID,
+  const treasuryAccount = "0x6d6f646c70792f74727372790000000000000000";
+  const mintingAmount = BN_THOUSAND.mul(BN_TEN.pow(new BN(18)));
+  const chain = "Ethereum";
```

Comment:
I think I did not explained myself,  PUT this on L26, so its available and a constant for all the tests, you do not need to re-declare on every test.

Reply:
Done